evalml/pipelines/utils.py

```diff
@@ -88,8 +94,23 @@ def _get_preprocessing_components(X, y, problem_type, estimator_class):
     if len(categorical_cols.columns) > 0 and estimator_class not in {CatBoostClassifier, CatBoostRegressor}:
         pp_components.append(OneHotEncoder)
 
+    if sampler_name is not None:
```

Comment:
We add this before the scaler since a) The scaler turns boolean inputs (like those we expect from the OHE) to numeric values (-1, 1), which means we would be unable to choose the proper columns for SMOTENC, and b) previous literature suggests that it makes sense to sample before normalizing/standardizing the data for the model.

Reply:
@dsherry the comment was here since the next component that is added to the pipeline would be the standard scaler. I added the oversampling component after the OHE but before the standard scaler, since we should do the scaling after any sampling